.github/workflows/ci.yml

```
@@ -1,128 +1,269 @@
+  # -----------------------------------------------------------------------------
+  # TIER 3: SECURITY & DEEP ANALYSIS (Slower, Comprehensive)
+  # -----------------------------------------------------------------------------
+  security-scans:
+    name: 🔒 Security Scans
+    needs: quality-checks
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - name: Run Trivy FS scan
+        with:
+          fetch-depth: 0 # Required for TruffleHog to scan history
```

YAMLlint (1.37.1)

[warning] too few spaces before comment: expected 2

(comments)